cirq-ionq/cirq_ionq/ionq_native_target_gateset_test.py

```diff
@@ -0,0 +1,712 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests transpiling of circuits using the IonQ native gate set."""
+
+import cirq
+import numpy as np
+import pytest
+
+from cirq_ionq.ionq_native_target_gateset import AriaNativeGateset
+from cirq_ionq.ionq_native_target_gateset import ForteNativeGateset
+
+gateset = AriaNativeGateset(atol=1e-8)
+
+
+# test object representation
+def test_AriaNativeGateset_repr():
+    gateset = AriaNativeGateset(atol=7)
+    assert repr(gateset) == 'cirq_ionq.AriaNativeGateset(atol=7)'
+
+
+# test object representation
+def test_ForteNativeGateset_repr():
+    gateset = ForteNativeGateset(atol=7)
+    assert repr(gateset) == 'cirq_ionq.ForteNativeGateset(atol=7)'
+
+
+# test init and atol argument
+def test_AriaNativeGateset_init():
+    gateset = AriaNativeGateset(atol=7)
+    assert gateset.atol == pytest.approx(7)
```

Comment:
not sure this is a good idea, we compare floating points numbers

Reply:
In this case we should use strict equality, because atol is assigned the argument value.
`pytest.approx` would be reasonable if atol was a result of some floating point operations that may introduce round off errors.  Assignment `a = b` does not do any flops and it must satisfy `a == b`.